clouddriver-cloudfoundry/src/test/java/com/netflix/spinnaker/clouddriver/cloudfoundry/deploy/ops/DestroyCloudFoundryServiceAtomicOperationTest.java

```diff
@@ -0,0 +1,100 @@
+import static com.netflix.spinnaker.clouddriver.cloudfoundry.client.model.v2.LastOperation.Type.DELETE;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.atIndex;
+import static org.mockito.Matchers.any;
+import static org.mockito.Mockito.when;
+
+class DestroyCloudFoundryServiceAtomicOperationTest extends AbstractCloudFoundryAtomicOperationTest {
+  private DestroyCloudFoundryServiceDescription desc = new DestroyCloudFoundryServiceDescription();
+
+  @Test
+  void destroyCloudFoundryService() {
+    desc.setServiceInstanceName("service-instance-name");
+    desc.setSpace(CloudFoundrySpace.builder().name("space-name").build());
+    desc.setClient(client);
+
+    ServiceInstanceResponse serviceInstanceResponse = new ServiceInstanceResponse()
+      .setServiceInstanceName("service-instance-name")
+      .setServiceInstanceId("service-instance-guid")
+      .setType(DELETE)
+      .setState(IN_PROGRESS);
+
+    when(client.getServiceInstances().destroyServiceInstance(any(), any()))
+      .thenReturn(serviceInstanceResponse);
+
+    DestroyCloudFoundryServiceAtomicOperation op = new DestroyCloudFoundryServiceAtomicOperation(desc);
+
+    Task task = runOperation(op);
+    List<Object> resultObjects = task.getResultObjects();
+    assertThat(1).isEqualTo(resultObjects.size());
+    Object o = resultObjects.get(0);
+    assertThat(o).isInstanceOf(ServiceInstanceResponse.class);
+    ServiceInstanceResponse response = (ServiceInstanceResponse) o;
+    assertThat(DELETE).isEqualTo(response.getType());
+    assertThat(IN_PROGRESS).isEqualTo(response.getState());
+    assertThat("service-instance-guid").isEqualTo(response.getServiceInstanceId());
+    assertThat("service-instance-name").isEqualTo(response.getServiceInstanceName());
+    assertThat(task.getHistory())
+      .has(status("Started removing service instance 'service-instance-name' from space space-name"), atIndex(1));
+  }
+
+  @Test
+  void destroyUserProvidedService() {
+    desc.setServiceInstanceName("up-service-instance-name");
+    desc.setSpace(CloudFoundrySpace.builder().name("space-name").build());
+    desc.setClient(client);
+
+    ServiceInstanceResponse serviceInstanceResponse = new ServiceInstanceResponse()
+      .setServiceInstanceName("up-service-instance-name")
+      .setServiceInstanceId("up-service-instance-guid")
+      .setType(DELETE)
+      .setState(NOT_FOUND);
+
+    when(client.getServiceInstances().destroyServiceInstance(any(), any()))
+      .thenReturn(serviceInstanceResponse);
+
+    DestroyCloudFoundryServiceAtomicOperation op = new DestroyCloudFoundryServiceAtomicOperation(desc);
+
+    Task task = runOperation(op);
+    List<Object> resultObjects = task.getResultObjects();
+    assertThat(1).isEqualTo(resultObjects.size());
+    Object o = resultObjects.get(0);
+    assertThat(o).isInstanceOf(ServiceInstanceResponse.class);
+    ServiceInstanceResponse response = (ServiceInstanceResponse) o;
```

Comment:
I don't see much value in asserting the individual fields of the response you created above. Maybe an equals against the whole object if you want to assert that your response is making it into the right place. I prefer tests that just have one (or very few asserts). In this case I guess the important one is the history? That just indirectly lets you know what happened though.

Reply:
👍